sdk/iot/core/src/az_iot_telemetry.c

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "az_iot_hub_client.h"
+#include <az_contract_internal.h>
+#include <az_result.h>
+#include <az_span.h>
+
+static az_span telemetry_prop_delim = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER((uint8_t[]){ '?' });
+static az_span telemetry_prop_separator
+    = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER((uint8_t[]){ '&' });
+static az_span telemetry_null_terminator
+    = AZ_SPAN_LITERAL_FROM_INITIALIZED_BUFFER((uint8_t[]){ '\0' });
+static az_span telemetry_topic_prefix = AZ_SPAN_LITERAL_FROM_STR("devices/");
+static az_span telemetry_topic_modules_mid = AZ_SPAN_LITERAL_FROM_STR("/modules/");
+static az_span telemetry_topic_suffix = AZ_SPAN_LITERAL_FROM_STR("/messages/events/");
+
+az_result az_iot_hub_client_telemetry_publish_topic_get(
+    az_iot_hub_client const* client,
```

Comment:
how about comments with example showing the topic example.

Reply:
I have added an example use of the API in the Readme. Is that what you are talking about?